posthog/rate_limit.py

```diff
@@ -0,0 +1,41 @@
+from django.urls.base import resolve
+from rest_framework.throttling import UserRateThrottle
+from sentry_sdk.api import capture_exception
+
+from posthog.internal_metrics import incr
+
+
+class PassThroughThrottle(UserRateThrottle):
+    def allow_request(self, request, view):
+        request_would_be_allowed = super().allow_request(request, view)
+        if not request_would_be_allowed:
+            try:
+                route = resolve(request.path)
+                route_id = f"{route.route} ({route.func.__name__})"
+                scope = getattr(self, "scope", None)
+                rate = self.get_rate() if scope else None
+                user_id = request.user.pk if request.user.is_authenticated else None
+                incr(
+                    "rate_limit_exceeded",
+                    tags={
```

Comment:
We can't add everything as tags - it would blow up influxdb pretty fast as every metric x unique tags combination gets stored as a separate series.

Reply:
🤔 for logs you'd want started throttling and stopped throttling rather than each throttled request, maybe

For grafana maybe URL path and team. So you can say which URLs are being throttle and for who

Reply:
Each of these tags should remain the same for each user/endpoint combo. If we're just catching kiddie scripts hitting a single endpoint with the rate limit, then that should be fine on influxdb. However, @macobo is right that if users end up hitting the rate limit from normal behavior (e.g. many endpoints), then it could blow up.

With that in mind, I just updated it to only tag the team + rate limit scope. This should give us enough information to begin tuning the limits, and when we see the volumes of rate limited users we can make a call to add more tags if needed.

Sound good?